app/models/item.rb

```diff
@@ -1,4 +1,13 @@
 class Item < ApplicationRecord
   belongs_to :wishlist
+  has_one :reservation, dependent: :destroy
   validates :name, presence: true
+
+  def reserved?
+    if reservation.nil?
+      I18n.t("gift_to_reserve")
+    else
+      I18n.t("gift_reserved")
+    end
+  end
```

Comment:
Nah. This is a model. Return `true` or `false`. There is no place to return parts of a view from a model. To return fancy strings create a Presenter that does that, or just put it directly in the view.